datafusion/expr/src/utils.rs

```diff
@@ -827,14 +827,13 @@ pub fn from_plan(
             window_expr: expr[0..window_expr.len()].to_vec(),
             schema: schema.clone(),
         })),
-        LogicalPlan::Aggregate(Aggregate {
-            group_expr, schema, ..
-        }) => Ok(LogicalPlan::Aggregate(Aggregate::try_new_with_schema(
-            Arc::new(inputs[0].clone()),
-            expr[0..group_expr.len()].to_vec(),
-            expr[group_expr.len()..].to_vec(),
-            schema.clone(),
-        )?)),
+        LogicalPlan::Aggregate(Aggregate { group_expr, .. }) => {
```

Comment:
I don't understand when this change would result in a different plan (the schema should always be unchanged if the same expressions are used)

Is this a change to make the code more defensive, or do you think it is causing a problem?

Related, perhaps we should remove `Aggregate::try_new_with_schema` as well (to force the recalculation / validation of the output schema)?

Reply:
> Related, perhaps we should remove Aggregate::try_new_with_schema as well (to force the recalculation / validation of the output schema)?

we don't need delete it, because some condition we don't change expr, such `with new inputs`

Reply:
Ah, yes, this makes sense now -- `new_with_plan` provide new (potentially different) exprs and inputs. Makes sense 👍